### ISSUE TITLE
Crash for windows386

### ISSUE THREAD

**Issue by Kvaz1r** [nelsam/vidar#87]
Today I was tried build vidar with go1.8.3 for windows under 386 arch and got a panic error when starting: 

> 2017/08/28 20:45:47 Warning: Error walking directory C:\: open C:\Documents and
> Settings: Access is denied.
> panic: runtime error: invalid memory address or nil pointer dereference
> [signal 0xc0000005 code=0x0 addr=0x0 pc=0x4019bb]
> 
> goroutine 19 [running]:
> sync/atomic.StoreUint64(0x13314324, 0x12, 0x0)
>         E:/go/src/sync/atomic/asm_386.s:190 +0xb
> github.com/nelsam/vidar/navigator.(*directory).reload(0x133141a0)
>         E:/gopath/src/github.com/nelsam/vidar/navigator/directory.go:132 +0x16f
> github.com/nelsam/vidar/navigator.newDirectory(0x12fd1130, 0x13101d9d, 0x3, 0x8)
> 
>         E:/gopath/src/github.com/nelsam/vidar/navigator/directory.go:75 +0x23b
> github.com/nelsam/vidar/navigator.(*ProjectTree).SetRoot(0x12fd1130, 0x13101d9d,
>  0x3)
>         E:/gopath/src/github.com/nelsam/vidar/navigator/project_tree.go:131 +0x1
> 22
> github.com/nelsam/vidar/navigator.(*ProjectTree).SetProject(0x12fd1130, 0x9c83fd
> , 0x9, 0x13101d9d, 0x3, 0x13101db0, 0x9)
>         E:/gopath/src/github.com/nelsam/vidar/navigator/project_tree.go:237 +0x3
> 1
> github.com/nelsam/vidar/navigator.NewProjectTree(0xcd2880, 0x1310cf60, 0xcd6ee0,
>  0x13122a00, 0x12ff4f00, 0xcd1ac0)
>         E:/gopath/src/github.com/nelsam/vidar/navigator/project_tree.go:95 +0x17
> a
> main.uiMain(0xcd2880, 0x1310cf60)
>         E:/gopath/src/github.com/nelsam/vidar/main.go:113 +0x23f
> github.com/nelsam/gxui/drivers/gl.StartDriver.func1()
>         E:/gopath/src/github.com/nelsam/gxui/drivers/gl/driver.go:57 +0x2e
> github.com/nelsam/gxui/drivers/gl.(*driver).applicationLoop(0x1310cf60)
>         E:/gopath/src/github.com/nelsam/gxui/drivers/gl/driver.go:111 +0x35
> created by github.com/nelsam/gxui/drivers/gl.StartDriver
>         E:/gopath/src/github.com/nelsam/gxui/drivers/gl/driver.go:58 +0x2d7

**Comment by Kvaz1r** [comment]
This really strange for me, but if permute the length field up: 
```
type directory struct {
	// length is an atomically updated list of child nodes of
	// this directory.  Only access via atomics.
	length int64
	mixins.LinearLayout

	driver gxui.Driver
	button *treeButton
	tree   *dirTree
}
```
vidar runs as should.

**Comment by nelsam** [comment]
That's a side effect of `int64` on 32-bit platforms.  From the [`atomic` docs](https://golang.org/pkg/sync/atomic/#pkg-note-BUG):

`On both ARM and x86-32, it is the caller's responsibility to arrange for 64-bit alignment of 64-bit words accessed atomically. The first word in a variable or in an allocated struct, array, or slice can be relied upon to be 64-bit aligned.`

If you feel like making a PR to move `length` to the beginning of the struct, please include a comment that explains that it *must* be first because of 32-bit platforms.  If you don't have time right now, I'll make the change myself in the coming week or two.

**Comment by nelsam** [comment]
This should be fixed in `master` now that #80 is merged.
